python/pyspark/ml/util.py

```diff
@@ -52,3 +71,141 @@ def _randomUID(cls):
         concatenates the class name, "_", and 12 random hex chars.
         """
         return cls.__name__ + "_" + uuid.uuid4().hex[12:]
+
+
+@inherit_doc
+class MLWriter(object):
+    """
+    Abstract class for utility classes that can save ML instances.
+
+    .. versionadded:: 2.0.0
+    """
+
+    def __init__(self, instance):
+        self._jwrite = instance._java_obj.write()
+
+    @since("2.0.0")
+    def save(self, path):
```

Comment:
This is the problematic workflow I had in mind:
```
lr = LinearRegression() // JVM object instantiated
lr.setMaxIter(10) // Python value set, but not transferred to JVM object
lr.save(...)  // new maxIter value is not saved
```

I think it will be easy to fix; just call ```_transfer_params_to_java``` in the init method of MLWriter.